lib/ansible/modules/clustering/k8s/k8s_service.py

```diff
@@ -0,0 +1,618 @@
+        u'spec': {}
+    }
+
+    template['metadata']['name'] = module.params['name']
+    if module.params.get('labels'):
+        template['metadata']['labels'] = module.params.get('labels')
+    if module.params.get('annotations'):
+        template['metadata']['annotations'] = module.params.get('annotations')
+    if module.params.get('cluster_ip'):
+        template['spec']['clusterIP'] = module.params.get('cluster_ip')
+    if module.params.get('external_i_ps'):
+        template['spec']['externalIPs'] = module.params.get('external_i_ps')
+    if module.params.get('type'):
+        template['spec']['type'] = module.params.get('type')
+    if module.params.get('external_name'):
+        template['spec']['externalName'] = module.params.get('external_name')
+    if module.params.get('load_balancer_source_ranges'):
+        template['spec']['loadBalancerSourceRanges'] = module.params.get('load_balancer_source_ranges')
+    if module.params.get('ports'):
+        template['spec']['ports'] = convert_port_names(module.params['ports'])
+    if module.params.get('selector'):
+        template['spec']['selector'] = module.params.get('selector')
+    if module.params.get('session_affinity'):
+        template['spec']['sessionAffinity'] = module.params.get('session_affinity')
+
+    api_response = template
+    if not module.check_mode:
+        try:
+            api_response = kube.create_namespaced_service(module.params['namespace'], template)
+        except ApiException as exc:
+            module.fail_json(msg="Exception when calling CoreV1Api->create_namespaced_service: {}".format(str(exc)))
+
+    return api_response
+
+
+def convert_port_names(ports):
+    '''
+    Convert parameter names to k8s template format
+    :param ports: list
+    :return: list
+    '''
+    template_ports = []
+    for port in ports:
+        template_port = {}
+        if port.get('name'):
+            template_port['name'] = port['name']
+        if port.get('protocol'):
+            template_port['protocol'] = port['protocol']
+        if port.get('port'):
+            template_port['port'] = port['port']
+        if port.get('target_port'):
+            template_port['targetPort'] = port['target_port']
+        if port.get('node_port'):
+            template_port['nodePort'] = port['node_port']
+        template_ports.append(template_port)
+    return template_ports
+
+
+def remove_service(module, kube):
+    '''
+    Delete an existing service.
+    :param module: Ansible module object
+    :param kube: kubernetes client
+    :param check_mode: bool
+    :return: None
+    '''
+
+    name = module.params['name']
+    namespace = module.params['namespace']
+
+    if not module.check_mode:
+        try:
+            kube.delete_namespaced_service(name, namespace)
+        except ApiException as exc:
+            module.fail_json(msg="Exception when calling CoreV1Api->delete_namespaced_service: {}".format(str(exc)))
+
+
+def patch_service(module, kube, service):
```

Comment:
This method seems very fragile, and I think we'd want to avoid this pattern as we look at expanding the set of modules...  It seems like we should be able to have a more generic method in module_utils for comparing and generating a patch (or updating an instantiation of a model based on the diff).